src/main/java/appeng/core/Registration.java

```diff
@@ -150,18 +143,11 @@ void preInitialize( final FMLPreInitializationEvent event )
 
 		MinecraftForge.EVENT_BUS.register( OreDictionaryHandler.INSTANCE );
 
-		final ApiDefinitions definitions = api.definitions();
+		ApiDefinitions definitions = api.definitions();
```

Comment:
It is one of the very few optimizations javac actually does. Because it is so trivial to do.

Iirc there are 4 optimization it does

1. constant arithmetic: 2+3+4 => 11
2. trivial boolean: `if(!foo) {} else {}` => `if(foo) {} else {}` or de morgan's laws
3. string concat: either just concat it or replace with a `StringBuilder` (might not be perfect)
4. trivial dead code elimination: like `if(false) {}`

It probably does not affect it much. But for example the old rendering did constantly use something like `int a = 65536; int b = a << 16 | a` or something like that. But at least it does not hurt.

> I just think final all over the place hurts readability.

A bit. But that is Java. With scala it would be easier as there is only `var foo, val bar, def baz`

No idea about hsdis. But I would not trust it. The JVM might only decide to optimize it after it has run for an hour. Just to throw the optimisation 5 secs later away and revert to the old version.